lua/typescript-tools/api.lua

```diff
@@ -161,6 +161,47 @@ function M.request_diagnostics(callback)
   }, callback)
 end
 
+---@param is_sync boolean
+function M.rename_file(is_sync)
+  local typescript_client = get_typescript_client(0)
+  if typescript_client == nil then
+    return
+  end
+
+  local source = vim.api.nvim_buf_get_name(0)
+
+  local params = {
+    files = {
+      {
+        oldUri = "file://" .. source,
+      },
+    },
+  }
+
+  vim.ui.input({ prompt = "New path: ", default = source }, function(input)
+    if input == "" or input == source or input == nil then
+      return
+    end
+
+    params.files[1].newUri = "file://" .. input
```

Comment:
You can use https://neovim.io/doc/user/lua.html#vim.uri_from_fname() to generate the uri.